File: 백준/Silver/1629. 곱셈/곱셈.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		int a = Integer.parseInt(st.nextToken());
		int b = Integer.parseInt(st.nextToken());
		int c = Integer.parseInt(st.nextToken());

		System.out.println(pow(a, b, c));
	}

	private static long pow(int a, int exponent, int mod) {

		if (exponent == 1) {
			return a % mod;
		}

		long temp = pow(a, exponent / 2, mod);

		if (exponent % 2 == 0) {
			return (temp * temp) % mod;
		} else {
			return (((temp * temp) % mod) * a) % mod;
		}
	}
}

File: 백준/Silver/1629. 곱셈/곱셈.py
import sys

a, b, c = map(int, sys.stdin.readline().rstrip().split())


def pow(a, exponent, mod):
    if exponent == 1:
        return a % mod

    temp = pow(a, exponent // 2, mod)

    if exponent % 2 == 0:
        return (temp**2) % mod
    else:
        return (((temp**2) % mod) * a) % mod


print(pow(a, b, c))
